File: packages/python/src/kimpton_evalrouter/_resources.py
"""Generated resource methods. Edit OpenAPI and client-operations.json."""

from __future__ import annotations

from typing import cast

from . import types
from ._boundary import validate_submission
from ._helpers import RunHelpers
from ._transport import RequestOptions, Resource, segment


class Catalog(Resource):
    def evals(
        self,
        *,
        params: types.CatalogEvalsParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.Page_EvalRecord_:
        return cast(
            types.Page_EvalRecord_,
            self._client.request("GET", "/evals", retry=True, options=options, query=params),
        )

    def eval(self, ref: str, *, options: RequestOptions | None = None) -> types.EvalRecord:
        return cast(
            types.EvalRecord,
            self._client.request("GET", f"/evals/{segment(ref)}", retry=True, options=options),
        )

    def benchmarks(
        self,
        *,
        params: types.CatalogBenchmarksParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.Page_BenchmarkRecord_:
        return cast(
            types.Page_BenchmarkRecord_,
            self._client.request(
                "GET", "/catalog/benchmarks", retry=True, options=options, query=params
            ),
        )

    def benchmark(
        self, slug: str, *, options: RequestOptions | None = None
    ) -> types.BenchmarkDetail:
        return cast(
            types.BenchmarkDetail,
            self._client.request(
                "GET", f"/catalog/benchmarks/{segment(slug)}", retry=True, options=options
            ),
        )

    def models(
        self,
        *,
        params: types.CatalogModelsParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.Page_ManagedRoute_:
        return cast(
            types.Page_ManagedRoute_,
            self._client.request(
                "GET", "/catalog/models", retry=True, options=options, query=params
            ),
        )

    def execution_options(self, *, options: RequestOptions | None = None) -> types.ExecutionOptions:
        return cast(
            types.ExecutionOptions,
            self._client.request("GET", "/catalog/execution-options", retry=True, options=options),
        )

    def environments(
        self,
        *,
        params: types.CatalogEnvironmentsParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.Page_EnvironmentRecord_:
        return cast(
            types.Page_EnvironmentRecord_,
            self._client.request(
                "GET", "/catalog/environments", retry=True, options=options, query=params
            ),
        )

    def environment(
        self, family: str, *, options: RequestOptions | None = None
    ) -> types.EnvironmentFamily:
        return cast(
            types.EnvironmentFamily,
            self._client.request(
                "GET", f"/catalog/environments/{segment(family)}", retry=True, options=options
            ),
        )


class Connections(Resource):
    def list(
        self,
        *,
        params: types.ConnectionsListParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.Page_ConnectionRecord_:
        return cast(
            types.Page_ConnectionRecord_,
            self._client.request("GET", "/connections", retry=True, options=options, query=params),
        )

    def create(
        self, body: types.NewConnection, *, options: RequestOptions | None = None
    ) -> types.ConnectionRecord:
        return cast(
            types.ConnectionRecord,
            self._client.request("POST", "/connections", retry=False, options=options, body=body),
        )

    def check(
        self, connection_id: str, *, options: RequestOptions | None = None
    ) -> types.ConnectionRecord:
        return cast(
            types.ConnectionRecord,
            self._client.request(
                "POST", f"/connections/{segment(connection_id)}/check", retry=False, options=options
            ),
        )

    def update(
        self,
        connection_id: str,
        body: types.UpdateConnection,
        *,
        options: RequestOptions | None = None,
    ) -> types.ConnectionRecord:
        return cast(
            types.ConnectionRecord,
            self._client.request(
                "PATCH",
                f"/connections/{segment(connection_id)}",
                retry=False,
                options=options,
                body=body,
            ),
        )

    def disable(
        self, connection_id: str, *, options: RequestOptions | None = None
    ) -> types.OKResponse:
        return cast(
            types.OKResponse,
            self._client.request(
                "DELETE", f"/connections/{segment(connection_id)}", retry=False, options=options
            ),
        )


class Quotes(Resource):
    def create(
        self, body: types.NewQuote, *, options: RequestOptions | None = None
    ) -> types.QuoteRecord:
        validate_submission("quotes", body)
        return cast(
            types.QuoteRecord,
            self._client.request("POST", "/quotes", retry=False, options=options, body=body),
        )

    def get(self, quote_id: str, *, options: RequestOptions | None = None) -> types.QuoteRecord:
        return cast(
            types.QuoteRecord,
            self._client.request(
                "GET", f"/quotes/{segment(quote_id)}", retry=True, options=options
            ),
        )


class Runs(RunHelpers):
    def create(
        self, body: types.NewRun, *, idempotency_key: str, options: RequestOptions | None = None
    ) -> types.RunRecord:
        return cast(
            types.RunRecord,
            self._client.request(
                "POST",
                "/runs",
                retry=True,
                options=options,
                body=body,
                idempotency_key=idempotency_key,
            ),
        )

    def list(
        self, *, params: types.RunsListParams | None = None, options: RequestOptions | None = None
    ) -> types.Page_RunRecord_:
        return cast(
            types.Page_RunRecord_,
            self._client.request("GET", "/runs", retry=True, options=options, query=params),
        )

    def get(self, run_id: str, *, options: RequestOptions | None = None) -> types.RunRecord:
        return cast(
            types.RunRecord,
            self._client.request("GET", f"/runs/{segment(run_id)}", retry=True, options=options),
        )

    def receipt(self, run_id: str, *, options: RequestOptions | None = None) -> types.RunReceipt:
        return cast(
            types.RunReceipt,
            self._client.request(
                "GET", f"/runs/{segment(run_id)}/receipt", retry=True, options=options
            ),
        )

    def executions(
        self,
        run_id: str,
        *,
        params: types.RunsExecutionsParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.SandboxExecutionPage:
        return cast(
            types.SandboxExecutionPage,
            self._client.request(
                "GET",
                f"/runs/{segment(run_id)}/executions",
                retry=True,
                options=options,
                query=params,
            ),
        )

    def events(
        self,
        run_id: str,
        *,
        params: types.RunsEventsParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.EventPage:
        return cast(
            types.EventPage,
            self._client.request(
                "GET", f"/runs/{segment(run_id)}/events", retry=True, options=options, query=params
            ),
        )

    def cancel(self, run_id: str, *, options: RequestOptions | None = None) -> types.RunRecord:
        return cast(
            types.RunRecord,
            self._client.request(
                "POST", f"/runs/{segment(run_id)}/cancel", retry=True, options=options
            ),
        )

    def results(
        self,
        run_id: str,
        *,
        params: types.RunsResultsParams | None = None,
        options: RequestOptions | None = None,
    ) -> types.ResultRecord:
        return cast(
            types.ResultRecord,
            self._client.request(
                "GET", f"/runs/{segment(run_id)}/results", retry=True, options=options, query=params
            ),
        )

    def export(
        self,
        run_id: str,
        *,
        params: types.RunsExportParams | None = None,
        options: RequestOptions | None = None,
    ) -> bytes:
        return cast(
            bytes,
            self._client.request(
                "GET",
                f"/runs/{segment(run_id)}/export",
                retry=True,
                options=options,
                query=params,
                binary=True,
            ),
        )


class Evals(Resource):
    def list(
        self, *, params: types.EvalsListParams | None = None, options: RequestOptions | None = None
    ) -> types.Page_EvalRecord_:
        return cast(
            types.Page_EvalRecord_,
            self._client.request("GET", "/evals", retry=True, options=options, query=params),
        )

    def get(self, ref: str, *, options: RequestOptions | None = None) -> types.EvalRecord:
        return cast(
            types.EvalRecord,
            self._client.request("GET", f"/evals/{segment(ref)}", retry=True, options=options),
        )


class Evaluations(Resource):
    def create(
        self,
        body: types.NewEvaluation,
        *,
        idempotency_key: str,
        options: RequestOptions | None = None,
    ) -> types.RunRecord:
        validate_submission("evaluations", body)
        return cast(
            types.RunRecord,
            self._client.request(
                "POST",
                "/evaluations",
                retry=True,
                options=options,
                body=body,
                idempotency_key=idempotency_key,
            ),
        )

    def get(self, evaluation_id: str, *, options: RequestOptions | None = None) -> types.RunRecord:
        return cast(
            types.RunRecord,
            self._client.request(
                "GET", f"/evaluations/{segment(evaluation_id)}", retry=True, options=options
            ),
        )

File: packages/typescript/src/resources.ts
// Generated resource methods. Edit OpenAPI and client-operations.json.
import type * as types from "./types.js";
import { Resource, segment, type RequestOptions } from "./transport.js";
import { RunHelpers } from "./helpers.js";
import { validateSubmission } from "./boundary.js";

export class Catalog extends Resource {
  async evals(params: types.CatalogEvalsParams = {}, options?: RequestOptions): Promise<types.Page_EvalRecord_> {
    return this.client.request<types.Page_EvalRecord_>("GET", `/evals`, { retry: true, options, query: params });
  }
  async eval(ref: string, options?: RequestOptions): Promise<types.EvalRecord> {
    return this.client.request<types.EvalRecord>("GET", `/evals/${segment(ref)}`, { retry: true, options });
  }
  async benchmarks(params: types.CatalogBenchmarksParams = {}, options?: RequestOptions): Promise<types.Page_BenchmarkRecord_> {
    return this.client.request<types.Page_BenchmarkRecord_>("GET", `/catalog/benchmarks`, { retry: true, options, query: params });
  }
  async benchmark(slug: string, options?: RequestOptions): Promise<types.BenchmarkDetail> {
    return this.client.request<types.BenchmarkDetail>("GET", `/catalog/benchmarks/${segment(slug)}`, { retry: true, options });
  }
  async models(params: types.CatalogModelsParams = {}, options?: RequestOptions): Promise<types.Page_ManagedRoute_> {
    return this.client.request<types.Page_ManagedRoute_>("GET", `/catalog/models`, { retry: true, options, query: params });
  }
  async execution_options(options?: RequestOptions): Promise<types.ExecutionOptions> {
    return this.client.request<types.ExecutionOptions>("GET", `/catalog/execution-options`, { retry: true, options });
  }
  async environments(params: types.CatalogEnvironmentsParams = {}, options?: RequestOptions): Promise<types.Page_EnvironmentRecord_> {
    return this.client.request<types.Page_EnvironmentRecord_>("GET", `/catalog/environments`, { retry: true, options, query: params });
  }
  async environment(family: string, options?: RequestOptions): Promise<types.EnvironmentFamily> {
    return this.client.request<types.EnvironmentFamily>("GET", `/catalog/environments/${segment(family)}`, { retry: true, options });
  }
}

export class Connections extends Resource {
  async list(params: types.ConnectionsListParams = {}, options?: RequestOptions): Promise<types.Page_ConnectionRecord_> {
    return this.client.request<types.Page_ConnectionRecord_>("GET", `/connections`, { retry: true, options, query: params });
  }
  async create(body: types.NewConnection, options?: RequestOptions): Promise<types.ConnectionRecord> {
    return this.client.request<types.ConnectionRecord>("POST", `/connections`, { retry: false, options, body: body });
  }
  async check(connection_id: string, options?: RequestOptions): Promise<types.ConnectionRecord> {
    return this.client.request<types.ConnectionRecord>("POST", `/connections/${segment(connection_id)}/check`, { retry: false, options });
  }
  async update(connection_id: string, body: types.UpdateConnection, options?: RequestOptions): Promise<types.ConnectionRecord> {
    return this.client.request<types.ConnectionRecord>("PATCH", `/connections/${segment(connection_id)}`, { retry: false, options, body: body });
  }
  async disable(connection_id: string, options?: RequestOptions): Promise<types.OKResponse> {
    return this.client.request<types.OKResponse>("DELETE", `/connections/${segment(connection_id)}`, { retry: false, options });
  }
}

export class Quotes extends Resource {
  async create(body: types.NewQuote, options?: RequestOptions): Promise<types.QuoteRecord> {
    validateSubmission("quotes", body);
    return this.client.request<types.QuoteRecord>("POST", `/quotes`, { retry: false, options, body: body });
  }
  async get(quote_id: string, options?: RequestOptions): Promise<types.QuoteRecord> {
    return this.client.request<types.QuoteRecord>("GET", `/quotes/${segment(quote_id)}`, { retry: true, options });
  }
}

export class Runs extends RunHelpers {
  async create(body: types.NewRun, idempotency_key: string, options?: RequestOptions): Promise<types.RunRecord> {
    return this.client.request<types.RunRecord>("POST", `/runs`, { retry: true, options, body: body, idempotency_key: idempotency_key });
  }
  async list(params: types.RunsListParams = {}, options?: RequestOptions): Promise<types.Page_RunRecord_> {
    return this.client.request<types.Page_RunRecord_>("GET", `/runs`, { retry: true, options, query: params });
  }
  async get(run_id: string, options?: RequestOptions): Promise<types.RunRecord> {
    return this.client.request<types.RunRecord>("GET", `/runs/${segment(run_id)}`, { retry: true, options });
  }
  async receipt(run_id: string, options?: RequestOptions): Promise<types.RunReceipt> {
    return this.client.request<types.RunReceipt>("GET", `/runs/${segment(run_id)}/receipt`, { retry: true, options });
  }
  async executions(run_id: string, params: types.RunsExecutionsParams = {}, options?: RequestOptions): Promise<types.SandboxExecutionPage> {
    return this.client.request<types.SandboxExecutionPage>("GET", `/runs/${segment(run_id)}/executions`, { retry: true, options, query: params });
  }
  async events(run_id: string, params: types.RunsEventsParams = {}, options?: RequestOptions): Promise<types.EventPage> {
    return this.client.request<types.EventPage>("GET", `/runs/${segment(run_id)}/events`, { retry: true, options, query: params });
  }
  async cancel(run_id: string, options?: RequestOptions): Promise<types.RunRecord> {
    return this.client.request<types.RunRecord>("POST", `/runs/${segment(run_id)}/cancel`, { retry: true, options });
  }
  async results(run_id: string, params: types.RunsResultsParams = {}, options?: RequestOptions): Promise<types.ResultRecord> {
    return this.client.request<types.ResultRecord>("GET", `/runs/${segment(run_id)}/results`, { retry: true, options, query: params });
  }
  async export(run_id: string, params: types.RunsExportParams = {}, options?: RequestOptions): Promise<Uint8Array> {
    return this.client.request<Uint8Array>("GET", `/runs/${segment(run_id)}/export`, { retry: true, options, query: params, binary: true });
  }
}

export class Evals extends Resource {
  async list(params: types.EvalsListParams = {}, options?: RequestOptions): Promise<types.Page_EvalRecord_> {
    return this.client.request<types.Page_EvalRecord_>("GET", `/evals`, { retry: true, options, query: params });
  }
  async get(ref: string, options?: RequestOptions): Promise<types.EvalRecord> {
    return this.client.request<types.EvalRecord>("GET", `/evals/${segment(ref)}`, { retry: true, options });
  }
}

export class Evaluations extends Resource {
  async create(body: types.NewEvaluation, idempotency_key: string, options?: RequestOptions): Promise<types.RunRecord> {
    validateSubmission("evaluations", body);
    return this.client.request<types.RunRecord>("POST", `/evaluations`, { retry: true, options, body: body, idempotency_key: idempotency_key });
  }
  async get(evaluation_id: string, options?: RequestOptions): Promise<types.RunRecord> {
    return this.client.request<types.RunRecord>("GET", `/evaluations/${segment(evaluation_id)}`, { retry: true, options });
  }
}
